opal/datatype/opal_datatype_memcpy.h

```diff
@@ -11,10 +11,45 @@
  * $HEADER$
  */
 
+#include "opal/runtime/opal.h"
+
 #ifndef OPAL_DATATYPE_MEMCPY_H_HAS_BEEN_INCLUDED
 #define OPAL_DATATYPE_MEMCPY_H_HAS_BEEN_INCLUDED
 
-#define MEMCPY( DST, SRC, BLENGTH ) \
-    memcpy( (DST), (SRC), (BLENGTH) )
+#define TYPED_MEMCPY( DST, SRC, BLENGTH, ALIGN, TYPE)                   \
+    if( ((BLENGTH) >= sizeof(TYPE)) && (0 == ((ALIGN) & (sizeof(TYPE) - 1))) ) { \
+        TYPE *__dst = (TYPE*)(DST), *__src = (TYPE*)(SRC);              \
+        if( (BLENGTH) == sizeof(TYPE) ) {                               \
+            *__dst = *__src;                                            \
+            break;                                                      \
+        }                                                               \
+        size_t _cnt = ((BLENGTH) / sizeof(TYPE));                       \
```

Comment:
Is it really worth doing all this instead of a memcpy() if blength > sizeof(type)?
I'll do some measurements...